helpers.py

```diff
@@ -0,0 +1,162 @@
+        time_stamp = session_date+session_time
+        baseline_temporal_freq = session_settings.temporalfreq
+        auto_rewd = session_settings.autorewd
+
+        for trial in np.arange(len(per_session_trials_data)):
+            # trial = 0 #comment out after loop structuring
+            all_mouse_session_dict['mouse_id'].append(mouse_id)
+            all_mouse_session_dict['time_stamp'].append(time_stamp)
+            trial_outcome = per_session_trials_data[trial]['trialoutcome']
+            all_mouse_session_dict['session_trial_outcome'].append(trial_outcome) # a single string value of FA, Hit, Ref, Miss, Gray, abort
+            if trial_outcome == 'Hit': trial_outcome = 'RT'
+            all_mouse_session_dict['reaction_time'].append(per_session_trials_data[trial]['reactiontimes'][trial_outcome]) # an individual float value from a dictionary of reaction times for each one of the keys for the possible trial outcomes detailed above. 
+            temporal_freq_vector = np.array(per_session_trials_data[trial]['TF']) # a list (turned into an array) of values for the temporal frequency from before grating started till the trial ended. 
+            temporal_freq_filter = temporal_freq_vector > 0
+            temporal_freq_vector = temporal_freq_vector[temporal_freq_filter]
+            temporal_freq_vector = temporal_freq_vector[::3]
+            all_mouse_session_dict['session_TF_vectors'].append(temporal_freq_vector)
+            all_mouse_session_dict['baseline_TF'].append(per_session_trials_data[trial]['Stim1TF']) # the value of TF around which the baseline frequency revolves. 
+            all_mouse_session_dict['change_size_TF'].append(per_session_trials_data[trial]['Stim2TF']) # the value of the change TF (change size out of 4,2,1.5,1.35,1.25)
+            all_mouse_session_dict['change_time'].append(per_session_trials_data[trial]['stimT'])
+            all_mouse_session_dict['auto_reward'].append(auto_rewd)
+    if return_dict_or_df != 'dict':
+        master_df = pd.DataFrame(all_mouse_session_dict)
+        try:
+            master_df['time_stamp'] = pd.to_datetime(master_df['time_stamp'],format='%Y%m%d%H%M%S')
+        except:
+            print('time stamp field of a file requires modification to convert to datetime')
+        master_df = master_df[(master_df['change_size_TF']>=1.25) & (master_df['change_size_TF']<15)]
+        if choose_outcome != 'all':
+            master_df = master_df[master_df['session_trial_outcome']==choose_outcome]
+        if include_auto_reward_trials == False:
+            master_df = master_df[master_df['auto_reward'] == 0]
+
+        master_df.reset_index(drop=True,inplace=True)
+
+        if return_dict_or_df == 'df': return master_df 
+        if return_dict_or_df == 'all': return all_mouse_session_dict, master_df
+        
+    # if return_dict_or_df := 'df': return master_df
+    
+    return all_mouse_session_dict
```

Comment:
In general it is good practice where possible to only return once from a function (it is not always possible). In this case I would consider only returning the `dict`, then having another high level function called something like `dict_to_df`. In `get_behav_data.py` if you do something like

```
behav_dict = gp.get_behave_data...
if return_dict_or_df == "df":
    behav_df = hp.get_behav_df(behav_dict )
```
It just makes the logic a little more explicit from a high level, and protects against future bugs if you refactor code where one function can have multiple output arguments.

Alternatively, if there is a clear reason to prefer one format over the other, I would stick with one that you will always use and return only that (as if I understand correctly they hold the same information, just different format)